tmva/tmva/inc/TMVA/DataLoader.h

```diff
@@ -226,7 +226,8 @@ namespace TMVA {
       ClassDef(DataLoader,3);
    };
    void DataLoaderCopy(TMVA::DataLoader* des, TMVA::DataLoader* src);
+   std::vector<std::shared_ptr<TFile>> DataLoaderCopyMP(TMVA::DataLoader *des, TMVA::DataLoader *src);
```

Comment:
Do these need to be public? Would it be better to use private with Envelope as a friend? I think the intention is that only the Envelope should be able to use it. I would then propose that we enforce this. We can always open the interface up later, but not the other way around.

Reply:
@mammadhajili could you answer the comments of @ashlaban please?